src/OpenTelemetry/Trace/ActivitySourceAdapter.cs

```diff
@@ -136,7 +136,7 @@ private void RunGetRequestedDataAlwaysOffSampler(Activity activity)
         private void RunGetRequestedDataOtherSampler(Activity activity)
         {
             ActivityContext parentContext;
-            if (string.IsNullOrEmpty(activity.ParentId))
+            if (string.IsNullOrEmpty(activity.ParentId) || this.IsParentSpanIdEmpty(activity))
```

Comment:
Lets add a comment to say something like:
Check activity.ParentId alone is sufficient to normally determine if a activity is root or not. But if one uses activity.SetParentId to override the TraceId, (without intending to set an actual parent), then additional check of parentspanid being empty is required to confirm if an activity is root or not.
This can be removed, once Activity exposes an API to customize ID Generation....

Reply:
Thanks, just added. Also, created an issue in the .NET runtime asking to expose the API to override the trace id.